Add descriptions to PaginationMetadataSchema

When upgrading a service to `flask-smorest@0.38.1`, it came with upgrading to `apispec@5.2.2`. Then, when running our openapi linting step, the following came up:

```
+ flask openapi write openapi.json     
+ spectral lint openapi.json --fail-severity=warn                                                                                                                
OpenAPI 3.x detected                                                                                                                                             
                                                                                                                                                                 
<...>/openapi.json                                                                                                                 
 2218:28  warning  oas3-unused-components-schema  Potentially unused components schema has been detected.  components.schemas.PaginationMetadata                 
 2221:19    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.total
 2224:25    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.total
_pages                                                                                                                                                           
 2227:24    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.first
_page                                                                                                                                                            
 2230:23    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.last_
page                                                                                                                                                             
 2233:18    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.page 
 2236:27    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.previ
ous_page                                                                                                                                                         
 2239:23    error  property-description           Property must have a description.                        components.schemas.PaginationMetadata.properties.next_
page                                                                                                                                                             
                                                                                                                                                                 
✖ 8 problems (7 errors, 1 warning, 0 infos, 0 hints)
```

I think the fix is simple and just requires adding a `metadata['description']` to the fields [here](https://github.com/marshmallow-code/flask-smorest/blob/master/flask_smorest/pagination.py#L112). This is also happening because we enforce having a description on our open api fields. Seems it would be a harmless update so I can put up a PR later probably, just wanted to log this and see what others think. I can also just ignore the error in the mean time :)

--- a/src/flask_smorest/pagination.py
+++ b/src/flask_smorest/pagination.py
@@ -118,13 +118,13 @@ class PaginationMetadataSchema(ma.Schema):
     Its main purpose is to document the pagination metadata.
     """
 
-    total = ma.fields.Int()
-    total_pages = ma.fields.Int()
-    first_page = ma.fields.Int()
-    last_page = ma.fields.Int()
-    page = ma.fields.Int()
-    previous_page = ma.fields.Int()
-    next_page = ma.fields.Int()
+    total = ma.fields.Int(metadata={"description": "Total number of items."})
+    total_pages = ma.fields.Int(metadata={"description": "Total number of pages."})
+    first_page = ma.fields.Int(metadata={"description": "First available page number."})
+    last_page = ma.fields.Int(metadata={"description": "Last available page number."})
+    page = ma.fields.Int(metadata={"description": "Current page number."})
+    previous_page = ma.fields.Int(metadata={"description": "Previous page number."})
+    next_page = ma.fields.Int(metadata={"description": "Next page number."})
 
 
 PAGINATION_HEADER = {

--- a/tests/test_pagination.py
+++ b/tests/test_pagination.py
@@ -193,6 +193,53 @@ class TestPagination:
                 }
             }
 
+    @pytest.mark.parametrize("openapi_version", ("2.0", "3.0.2"))
+    def test_pagination_metadata_documentation(self, app, openapi_version):
+        """Test pagination metadata fields are documented"""
+        app.config["OPENAPI_VERSION"] = openapi_version
+        api = Api(app)
+        blp = Blueprint("test", __name__, url_prefix="/test")
+
+        @blp.route("/")
+        @blp.response(200)
+        @blp.paginate()
+        def func(pagination_parameters):
+            """Dummy view func"""
+
+        api.register_blueprint(blp)
+        pagination_metadata = get_schemas(api.spec)["PaginationMetadata"]
+
+        assert pagination_metadata["properties"] == {
+            "total": {
+                "type": "integer",
+                "description": "Total number of items.",
+            },
+            "total_pages": {
+                "type": "integer",
+                "description": "Total number of pages.",
+            },
+            "first_page": {
+                "type": "integer",
+                "description": "First available page number.",
+            },
+            "last_page": {
+                "type": "integer",
+                "description": "Last available page number.",
+            },
+            "page": {
+                "type": "integer",
+                "description": "Current page number.",
+            },
+            "previous_page": {
+                "type": "integer",
+                "description": "Previous page number.",
+            },
+            "next_page": {
+                "type": "integer",
+                "description": "Next page number.",
+            },
+        }
+
     @pytest.mark.parametrize("header_name", ("X-Pagination", None))
     def test_pagination_item_count_missing(self, app, header_name, recwarn):
         """If item_count was not set, pass and warn"""

